File: old_version/Java/第5章 树/5.1 AVL树/AVLTest.java
public class AVLTest {
    public static void main(String[] args) {
        int[] arr = {3, 2, 1, 4, 5, 6, 7, 16, 15, 14, 13, 12, 11, 10, 8, 9};
        AVLTree<Integer> avl = new AVLTree<>();
        for(int i = 0; i < arr.length; i++) {
            avl.insert(arr[i]);
        }

        System.out.print("preOrder: ");
        avl.preOrder();
        System.out.println();

        System.out.print("inOrder: ");
        avl.inOrder();
        System.out.println();

        System.out.print("postOrder: ");
        avl.postOrder();
        System.out.println();

        avl.printTree();
    }
}

File: code/Python/第17章 树/17.1 AVL树/avl_test.py
import avl_tree

def main():
    lst = [3, 2, 1, 4, 5, 6, 7, 16, 15, 14, 13, 12, 11, 10, 8, 9]
    avl = avl_tree.AVLTree()
    for data in lst:
        avl.insert(data)
    
    print("preOrder: ", end='')
    avl.pre_order()
    print()

    print("inOrder: ", end='')
    avl.in_order()
    print()

    print("poseOrder: ", end='')
    avl.post_order()
    print()

    avl.print_tree()

if __name__ == "__main__":
    main()